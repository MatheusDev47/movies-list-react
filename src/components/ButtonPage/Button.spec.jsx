import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import { ButtonPage } from "."

const fn = jest.fn()

describe('<ButtonPage />', () => {
    it('Should render the button with the text "Load more films"', () => {
        render(<ButtonPage text="Load more films"/>)

        expect(screen.getByRole('button', {name: /load more films/i})).toBeInTheDocument()
    })

    it('Should button function onClick is working', () => {
        render(<ButtonPage text="Load more films" onClick={fn} />)

        userEvent.click(screen.getByRole('button', { name: /load more films/i }))

        expect(fn).toHaveBeenCalledTimes(1)
    })

    it('Should button is disabled when disabled is true', () => {
        render(<ButtonPage text="Load more films" onClick={fn} disabled={true}/>)

        expect(screen.getByRole('button', { name: /load more films/i })).toBeDisabled()
    })

    it('Should button is enabled when disabled is false', () => {
        render(<ButtonPage text="Load more films" onClick={fn} disabled={false}/>)

        expect(screen.getByRole('button', { name: /load more films/i })).toBeEnabled()
    })

    it('Should match snapshot', () => {
        render(<ButtonPage text="Load more films" onClick={fn} disabled={false} />)

        expect(screen.getByRole('button', { name: /load more films/i })).toMatchSnapshot()
    })
})