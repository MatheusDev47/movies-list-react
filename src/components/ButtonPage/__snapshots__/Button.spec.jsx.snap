// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonPage /> Should match snapshot 1`] = `
<button
  class="btn-next"
>
  Load more films
</button>
`;
